pkg/substitution/substitution.go

```diff
@@ -366,3 +416,13 @@ func ExtractIndex(s string) (int, error) {
 func StripStarVarSubExpression(s string) string {
 	return strings.TrimSuffix(strings.TrimSuffix(strings.TrimPrefix(s, "$("), ")"), "[*]")
 }
+
+// parseIndex parses integer from a string.
+// If the string is not numeric, the second returned value will be false.
+func parseIndex(s string) (int, bool) {
```

Comment:
Wondering if we should stick with strconv.Atoi here? This func seems to be triming the error message to boolean.

Reply:
And also this func doesn't seem to be used elsewhere at the moment?